dali/operators/segmentation/random_mask_pixel.cc

```diff
@@ -0,0 +1,203 @@
+  explicit RandomMaskPixelCPU(const OpSpec &spec);
+  bool CanInferOutputs() const override { return true; }
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<CPUBackend> &ws) override;
+  void RunImpl(workspace_t<CPUBackend> &ws) override;
+
+ private:
+  template <typename T>
+  void RunImplTyped(workspace_t<CPUBackend> &ws);
+
+  int64_t seed_;
+  std::vector<std::mt19937_64> rng_;
+  std::vector<SearchableRLEMask> rle_;
+
+  std::vector<int> foreground_;
+  std::vector<int> value_;
+  std::vector<float> threshold_;
+
+  bool has_value_ = false;
+
+  USE_OPERATOR_MEMBERS();
+};
+
+RandomMaskPixelCPU::RandomMaskPixelCPU(const OpSpec &spec)
+    : Operator<CPUBackend>(spec),
+      seed_(spec.GetArgument<int64_t>("seed")),
+      has_value_(spec.ArgumentDefined("value")) {
+  if (has_value_) {
+    DALI_ENFORCE(!spec.ArgumentDefined("threshold"),
+      "Arguments ``value`` and ``threshold`` can not be provided together");
+  }
+}
+
+bool RandomMaskPixelCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                    const workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  output_desc.resize(1);
+  output_desc[0].shape = uniform_list_shape(nsamples, {ndim});
+  output_desc[0].type = TypeTable::GetTypeInfo(DALI_INT64);
+
+  foreground_.resize(nsamples);
+  value_.clear();
+  threshold_.clear();
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    foreground_[sample_idx] = spec_.template GetArgument<int>("foreground", &ws, sample_idx);
+  }
+
+  if (spec_.ArgumentDefined("value")) {
+    value_.resize(nsamples);
+    for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+      value_[sample_idx] = spec_.template GetArgument<int>("value", &ws, sample_idx);
+    }
+  } else {
+    threshold_.resize(nsamples, 0.0f);
+    if (spec_.ArgumentDefined("threshold")) {
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        threshold_[sample_idx] = spec_.template GetArgument<float>("threshold", &ws, sample_idx);
+      }
+    }
+  }
+  return true;
+}
+
+template <typename T>
+void RandomMaskPixelCPU::RunImplTyped(workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  auto &out_pixel_pos = ws.template OutputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  auto masks_view = view<const T>(in_masks);
+  auto pixel_pos_view = view<int64_t>(out_pixel_pos);
+  auto& thread_pool = ws.GetThreadPool();
+
+  if (rng_.empty()) {
+    for (int i = 0; i < thread_pool.size(); i++) {
+      rng_.emplace_back(seed_ + i);
+    }
+  }
+  assert(rng_.size() == static_cast<size_t>(thread_pool.size()));
+
+  if (rle_.empty()) {
+    rle_.resize(thread_pool.size());
+  }
```

Comment:
Nitpick: resize won't do much more than a similar check, so:
```suggestion
  rle_.resize(thread_pool.size());
```